src/map/channel.cpp

```diff
@@ -27,6 +27,227 @@ struct Channel_Config channel_config;
 
 DBMap* channel_get_db(void){ return channel_db; }
 
+const std::string ChannelConfigLoader::getDefaultLocation() {
+	return std::string(conf_path) + "/channels.yml";
+}
+
+bool ChannelConfigLoader::parseBody(const ryml::NodeRef& body) {
+	if (this->nodeExists(body, "Private")) {
+		parsePrivate(body["Private"]);
+	}
+
+	if (this->nodeExists(body, "Public")) {
+		for (const auto &node : body["Public"]) {
+			parsePublicNode(node);
+		}
+	}
+	return true;
+}
+
+
+bool ChannelConfigLoader::parsePrivate(const ryml::NodeRef& node) {
+
+	auto &conf = channel_config.private_channel;
+
+	if (nodeExists(node, "Allow")) {
+		bool active;
+		if (!asBool(node, "Allow", active))
+			return false;
+
+		conf.allow = active;
+	}
+
+	if (nodeExists(node, "Color")) {
+		uint32 color;
+		if (!asUInt32(node, "Color", color))
+			return false;
+		if (color > 0xFF'FF'FF) {
+			invalidWarning(node["Color"], "Invalid private channel color 0x%06x, defaulting to 0xFFFFFF.\n", color);
+			color = 0xFF'FF'FF;
```

Comment:
Maybe use clif.hpp::color_table[COLOR_DEFAULT] instead?
But since this is already in BGR, please make sure to put the line below into the else case then.